File: Programmers/Python3/Level 2/이진 변환 반복하기/solution.py
# https://school.programmers.co.kr/learn/courses/30/lessons/70129


def solution(s):
    convert, zero = 0, 0
    while s != "1":
        convert += 1
        one = 0
        for num in s:
            if num == "1":
                one += 1
            else:
                zero += 1
        s = str(bin(one)[2:])   # 앞에있는 0b를 제거하기위해 슬라이스 사용
    return [convert, zero]


print(solution("110010101001"))

File: Programmers/Javascript/Level 2/이진 변환 반복하기/solution.js
// https://school.programmers.co.kr/learn/courses/30/lessons/70129?language=javascript

function solution(s) {
  const answer = [0, 0];

  while (s.length > 1) {
    let zero = 0,
      one = 0;

    for (let idx = 0; idx < s.length; idx++) {
      if (s[idx] === "1") {
        one += 1;
      } else {
        zero += 1;
      }
    }

    s = (s.length - zero).toString(2);
    answer[1] += zero;
    answer[0] += 1;
  }
  return answer;
}

solution("110010101001");
